src/SignalR/common/Http.Connections/src/NegotiateMetadata.cs

```diff
@@ -0,0 +1,13 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.AspNetCore.Http.Connections
+{
+    /// <summary>
+    /// Metadata to identify the /negotiate endpoint for HTTP dispatcher based
+    /// </summary>
+    public class NegotiateMetadata
+    {
```

Comment:
@rynowak In routing we had interfaces for some metadata. Is there a rule of thumb for interface vs class?

https://github.com/aspnet/AspNetCore/blob/f10635f6c832641e022a05c7fd97fe512ec844d9/src/Http/Routing/src/ISuppressLinkGenerationMetadata.cs

Reply:
I'd like to merge this today sooo, we should totally bikeshed on this as fast as possible. I only made it a class because I was thinking we would add things to it but maybe it could be more metadata. Either way I don't super care.

Reply:
👍 	for the interface. I would say it's to be more similar to ISuppressLinkGenerationMetadata, but I don't see the need for a bool. The absence or presence of INegotiateMetadata should be sufficient.

Reply:
The presence or absence isn't enough as **a general rule** because metadata or ordered and can override.  On a case-by-case basis it can be fine 😆

Reply:
Interfaces are for cases where you might want to compose the metadata with some other type that implements multiple policies.

Also to allow you to use attributes to specify metadata without the metadata being tied to an attribute class. (`IAuthorizeData` vs `AuthorizeAttribute`).

Depending on the user case a class could be fine. Do you plan to have one ever one of these? Does it make sense for someone to customize, extend or replace it?

For instance the MVC `ActionDescriptor` goes in metadata but we have no plans to turn that into an interface.